### ISSUE TITLE
Inferred implicit any accessor type in object literal

### ISSUE THREAD

**Issue by kitsonk** [Microsoft/TypeScript#8373]
**TypeScript Version:** 

nightly (1.9.0-dev.20160429)

**Code**

```ts
const foo = {
	_foo: 'bar',
	get foo() { // <- 'foo' implicitly has return type 'any' because it does not have a return type annotation and is referenced directly or indirectly in one of its return expressions.
		return this._foo;
	},
	set foo(value) { // <- 'value' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.
		this._foo = value;
	}
};
```

**Expected behavior:**

No errors.

**Actual behavior:** 

With the compiler option `noImplicitAny = true` and inferred object literal typing, the compiler is complaining that `get` and `set` accessors have inferred `any` types, when though it is logical that the type of `string` can be inferred from the context (or at least I am missing the reason why it cannot be inferred).

**Comment by sandersn** [comment]
It doesn't repro for me on master. I thought the nightly from 04/29 would have the commits to revert object literal `this` typing to always be `any`, but maybe it doesn't. Let me figure out which commit 1.9.0-dev.20160429 is built from.

It does have errors all over the place with `--noImplicitThis`. Could that be what you're using?

**Comment by kitsonk** [comment]
I found it... `--strictNullChecks` on for `1.9.0-dev.20160429` it will not cause the behaviour.  If you turn `--strictNullChecks` off on `1.9.0-dev.20160429` it will manifest itself, but isn't present at all (either configuration) on `1.9.0-dev.20160503`.  Not sure what broke/fixed it... 😄

**Comment by sandersn** [comment]
I didn't have time to check the commit that the 20160429 nightly used, but it's probably the change I made that reverts `this` in object literals to `any`. I guess it went in too late to get picked up by the nightly build.
